scripts/samples/bench_and_plot.py

```diff
@@ -0,0 +1,354 @@
+    args.add_argument('--truncate_highest_frac', type=float, default=0.0001)
+    args.add_argument('--pad_lowest_frac', type=float, default=0.01)
+    args.add_argument('--plot_bins', type=int, default=500)
+    args.add_argument('--dpi', type=int, default=600)
+    return args.parse_args()
+
+
+@contextmanager
+def timed(text: str) -> Iterator:
+    """Time some activity within a contextmanager, logging to stdout.
+
+    Args:
+        text (str): The name of the activity to log.
+
+    Returns:
+        Iterator: The yield.
+    """
+    t0 = time()
+    yield
+    t = time() - t0
+    print(f'{text}: {t:.3f} s')
+
+
+def generate_text(dataset_size: int, text_len: int) -> Tuple[Dict[str, str], List[Dict[str, Any]]]:
+    """Generate dataset of text samples.
+
+    Args:
+        dataset_size (int): Number of samples.
+        text_len (int): Length of each text field.
+
+    Returns:
+        Tuple[Dict[str, str], List[Dict[str, Any]]]: Column names/types and each sample.
+    """
+    columns = {'text': 'str'}
+    vocab = np.array(list(map(ord, string.ascii_letters)))
+    assert (vocab < 128).all()
+    vocab = vocab.astype(np.uint8)
+    arr = np.random.choice(vocab, (dataset_size, text_len))
+    samples = []
+    for sample_id in trange(dataset_size, leave=False):
+        text = arr[sample_id].tobytes().decode('utf-8')
+        sample = {'text': text}
+        samples.append(sample)
+    return columns, samples
+
+
+def generate_tokens(dataset_size: int, tokens_per_sample: int, dtype: DTypeLike) -> \
+        Tuple[Dict[str, str], List[Dict[str, Any]]]:
+    """Generate dataset of token samples.
+
+    Args:
+        dataset_size (int): Number of samples.
+        tokens_per_sample (int): Length of each text field.
+        dtype (DTypeLike): Which dtype to use for the tokens ndarray.
+
+    Returns:
+        Tuple[Dict[str, str], List[Dict[str, Any]]]: Column names/types and each sample.
+    """
+    dtype = np.dtype(dtype)
+    columns = {'tokens': f'ndarray:{dtype.name}:{tokens_per_sample}'}
+    arr = np.random.randint(0, np.iinfo(dtype).max, (dataset_size, tokens_per_sample), dtype=dtype)
+    samples = []
+    for sample_id in trange(dataset_size, leave=False):
+        tokens = arr[sample_id]
+        sample = {'tokens': tokens}
+        samples.append(sample)
+    return columns, samples
+
+
+def write(writer_class: type, dirname: str, columns: Dict[str, str], size_limit: int,
+          samples: List[Dict[str, Any]]) -> None:
+    """Given a writer class and information to write, serialize a dataset in that format.
+
+    Args:
+        writer_class (type): Which writer to use (MDS, JSONL, etc.).
+        dirname (str): Dataset directory.
+        columns (Dict[str, str]): Dataset columns.
+        size_limit (int): Maximum shard size.
+        samples (List[Dict[str, str]]): Samples that will comprise the dataset.
+    """
+    if os.path.exists(dirname):
+        rmtree(dirname)
```

Comment:
nit: do we need `ignore_errors=True` ? For example, `rmtree(local, ignore_errors=True)`